packages/server/src/internals/graphql/utils.ts

```diff
@@ -118,3 +120,18 @@ export function assertStringIsNotEmpty(args: { [key: string]: unknown }) {
     )
   }
 }
+
+export function checkRoleInstanceCompatibility(
+  role: Role,
+  instance: Instance | null
+) {
+  if ([Role.Guest, Role.Login, Role.Sysadmin].includes(role)) {
+    if (isInstance(instance)) {
```

Comment:
```suggestion
  if ([Role.Guest, Role.Login, Role.Sysadmin].includes(role) && isInstance(instance)) {
```
that way you avoid a nested if. I prefer that way, but to be honest I don't think it makes it more easy to read 😆